datastore/importers/nmdb-import-systeminfo/Parser.test.cpp

```diff
@@ -0,0 +1,253 @@
+          Registered Owner:    Your Name
+          Registered Organization: 
+          Product ID:          XXXXX-XXXXX-XXXXX-XXXXX
+          Original Install Date: 01/01/2022, 12:00:00 AM
+          System Boot Time:    11/22/2023, 9:00:00 AM
+          System Manufacturer: Your Computer Manufacturer
+          System Model:        Your Computer Model
+          System Type:         x64-based PC
+          Processor(s):        1 Processor(s) Installed.
+                                [01]: Intel64 Family 6 Model 142 Stepping 9 GenuineIntel ~2.30 GHz
+          BIOS Version:        Your BIOS Version
+          Windows Directory:   C:\Windows
+          System Directory:    C:\Windows\system32
+          Boot Device:         \Device\HarddiskVolume1
+          System Locale:       en-us;English (United States)
+          Input Locale:        en-us;English (United States)
+          Time Zone:           (UTC-08:00) Pacific Time (US & Canada)
+          Total Physical Memory: 16,384 MB
+          Available Physical Memory: 8,192 MB
+          Virtual Memory: Max Size: 32,768 MB
+          Virtual Memory: Available: 20,480 MB
+          Virtual Memory: In Use: 12,288 MB
+          Page File Location(s): C:\pagefile.sys
+          Domain:              WORKGROUP
+          Logon Server:        \\YourLogonServer
+          Hotfix(s):           10 Hotfix(s) Installed.
+                                [01]: KBXXXXXX
+                                [02]: KBXXXXXX
+                                [03]: KBXXXXXX
+                                [04]: KBXXXXXX
+                                [05]: KBXXXXXX
+                                [06]: KBXXXXXX
+                                [07]: KBXXXXXX
+                                [08]: KBXXXXXX
+                                [09]: KBXXXXXX
+                                [10]: KBXXXXXX
+          Network Card(s):     2 NIC(s) Installed.
+                                [01]: Realtek PCIe GbE Family Controller
+                                  Connection Name: Ethernet
+                                  DHCP Enabled:    Yes
+                                  IP address(es)
+                                  [01]: 192.168.0.2
+                                  [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+                                [02]: Intel Wireless-AC 9560 160MHz
+                                  Connection Name: Wi-Fi
+                                  DHCP Enabled:    Yes
+                                  IP address(es)
+                                  [01]: 192.168.0.3
+                                  [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+          Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
+
+          )STR"
+   };
+
+  for (const auto& test : testsOk) {
+    BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
+              "Parse rule 'start': " << test);
+  }
```

Comment:
If you check state change validity at each rule, you shouldn't really need it here.  You can put it if you want, but it can get complicated if you have multiple formats you want to check.